torch/autograd/gradcheck.py

```diff
@@ -102,76 +114,166 @@ def get_stride(size):
 
 
-def compute_gradient(fn, inputs, x, idx, delta, eps, is_mkldnn):
+def _get_numerical_jacobian(fn, inputs, outputs=None, target=None, eps=1e-3,
+                            grad_out=1.0) -> List[Tuple[torch.Tensor, ...]]:
+    """Computes the numerical jacobian for a given fn and inputs. Returns M * N jacobians
+    where M is the number of input tensors that require grad, and N is the number of output
+    float/complex tensors.
+
+    Args:
+        fn: the function to compute the jacobian for
+        inputs: inputs to `fn`
+        outputs: provide precomputed outputs to avoid one extra invocation of fn
+        target: the Tensors wrt whom Jacobians are calculated (default=`inputs`)
+        eps: the magnitude of the perturbation during finite differencing (default=`1e-3`)
+        grad_out: grad output value used to calculate gradients.
+
+    Returns:
+        A list of M N-tuples of tensors
+
+    Note that `target` may not even be part of `input` to `fn`, so please be
+    **very careful** in this to not clone `target`.
+    """
+    jacobians: List[Tuple[torch.Tensor, ...]] = []
+    if outputs is None:
+        outputs = _as_tuple(fn(*_as_tuple(inputs)))
+    if target is None:
+        target = inputs
+    inp_indices = [i for i, a in enumerate(target) if is_tensor_like(a) and a.requires_grad]
+    for i, (inp, inp_idx) in enumerate(zip(iter_tensors(target, True), inp_indices)):
+        jacobians += [get_numerical_jacobian_helper(fn, inp, inp_idx, inputs, outputs, eps, grad_out)]
+    return jacobians
+
+
+def get_numerical_jacobian(fn, inputs, target=None, eps=1e-3, grad_out=1.0):
+    """Deprecated api to compute numerical jacobian for a given fn and inputs.
+    Args:
+        fn: the function to compute the jacobian for (must take inputs as a tuple)
+        input: input to `fn`
+        target: the Tensors wrt whom Jacobians are calculated (default=`input`)
+        eps: the magnitude of the perturbation during finite differencing (default=`1e-3`)
+        grad_out: grad output value used to calculate gradients.
+
+    Returns:
+        A list of jacobians wrt each input (or target) and the first output
+
+    Note that `target` may not even be part of `input` to `fn`, so please be
+    **very careful** in this to not clone `target`.
+    """
+    warnings.warn("get_numerical_jacobian was part of PyTorch's private API and not "
+                  "meant to be exposed. We are deprecating it and it will be removed "
+                  "in a future version of PyTorch. If you have a specific use for "
+                  "this or feature request for this to be a stable API, please file "
+                  "us an issue at https://github.com/pytorch/pytorch/issues/new")
+
+    def fn_pack_inps(*inps):
+        return fn(inps)
+    jacobians = _get_numerical_jacobian(fn_pack_inps, inputs, None, target, eps, grad_out)
+
+    return tuple(jacobian_for_each_output[0] for jacobian_for_each_output in jacobians)
```

Comment:
Updated this here: since old API always returned a tuple of tensors. Tensors represent jacobians wrt to each input and to a single output.